modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -559,11 +559,24 @@ public MetricManager metricManager() {
     private static void validateParsedStatement(
             QueryContext context,
             ParsedResult parsedResult,
+            @Nullable InternalTransaction outerTx,
             Object[] params
     ) {
         Set<SqlQueryType> allowedTypes = context.allowedQueryTypes();
         SqlQueryType queryType = parsedResult.queryType();
 
+        if (parsedResult.queryType() == SqlQueryType.TX_CONTROL) {
+            if (outerTx != null) {
```

Comment:
>Checking for an explicit transaction seems useless, because the user is currently prohibited from executing TX_CONTROL in any way

Control statement check can be moved to the caller method because `validateParsedStatement` is going to be used by execute script method.

Reply:
Fixed.